### DISCUSSION TITLE
Scratchpad get last window instead of first.

### DISCUSSION THREAD

**Discussion by geril07** [i3/i3#5693] [Q&A] [answered]
If i add to scratchpad `win1` , `win2` using `move scratchpad`, `scratchpad show` gives me `win1` which is the first window there. Can i get `win2` which is the last window by executing one command or some script?

**Comment by okraits** [comment]
Sure, use something like that:

`[class="win2"] scratchpad show`

It is explained in the userguide: https://i3wm.org/docs/userguide.html#command_criteria

**Reply by geril07** [reply]
In this way, i need to make 2 steps:
1. Find the last window in scratchpad and get some attribute(`class`).
2. Show window by attribute


How can i achieve 1 step?

**Reply by okraits** [reply]
I read your question like you wanted to get the `win2` window. You can use the IPC interface to get a list of all scratchpad windows and then get the last one.

**Comment by geril07** [comment, marked as answer]
Found solution, thanks.
Maybe someone needed 
https://github.com/geril2207/dotfiles/blob/main/linux/i3/bin/get_last_scratchpad
